sdv/constraints/base.py

```diff
@@ -323,9 +324,9 @@ def fit(self, table_data):
                 Table data.
         """
         data_to_model = table_data[self.constraint_columns]
-        self._hyper_transformer = HyperTransformer(
-            default_data_type_transformers={'categorical': 'OneHotEncodingTransformer'}
-        )
+        self._hyper_transformer = HyperTransformer()
+        self._hyper_transformer.detect_initial_config(data_to_model)
```

Comment:
Is it ok that we are detecting the config here? I know that's probably what we were doing previously, but I'm confused about how that works.

Reply:
I think that it is ok. The reason is that mainly we get the `data_to_model` which is a subset of the dataframe, we then `fit_transform` only that portion of the data.